addons/knobs/src/components/Panel.js

```diff
@@ -72,23 +72,22 @@ export default class KnobPanel extends PureComponent {
 
   setKnobs = ({ knobs, timestamp }) => {
     const queryParams = {};
-    const { api, channel } = this.props;
+    const { api } = this.props;
 
     if (!this.options.timestamps || !timestamp || this.lastEdit <= timestamp) {
       Object.keys(knobs).forEach(name => {
         const knob = knobs[name];
         // For the first time, get values from the URL and set them.
         if (!this.loadedFromUrl) {
           const urlValue = api.getQueryParam(`knob-${name}`);
+
+          // If the knob value present in url
           if (urlValue !== undefined) {
-            // If the knob value present in url
-            knob.value = Types[knob.type].deserialize(urlValue);
-            channel.emit(CHANGE, knob);
```

Comment:
Should we still fire an event?! I.e. `api.emit(CHANGE, knob)`; otherwise, this may be a minor breaking change for some people (-> can not cherry picked this one before v5.1).

Reply:
The event will be redundant, the knob manager will already be in that exact state.

I can fire the event, to be sure, but I don't see the point? What usage is this breaking?

Reply:
Ow.. 
> Minor concerns on RN

that probably?

Reply:
I think SB using events heavily... although in theory if the event is name spaced, it should be no problem at all because it is meant to be used internally.  But since SB is also highly customizable, so standing the user point, I will take more conservative approach. At least in the doc level... Don't won't freak out people if they have customized code working well for long time but fail without any notice... (so I say "may" be a breaking change for some users :p)

Reply:
Well not anymore, I added it back, be it with the `api.on` method, but that's all the same.